apps/platform/trpc/routers/authRouter/passkeyRouter.ts

```diff
@@ -7,58 +7,165 @@ import {
   RegistrationResponseJSON,
   AuthenticationResponseJSON
 } from '@simplewebauthn/types';
-import { nanoId } from '@uninbox/utils';
+import { nanoId, zodSchemas } from '@uninbox/utils';
 import { UAParser } from 'ua-parser-js';
 import { usePasskeys } from '../../../utils/auth/passkeys';
 import { usePasskeysDb } from '../../../utils/auth/passkeyDbAdaptor';
 import { setCookie, getCookie, getHeader } from '#imports';
 import { lucia } from '../../../utils/auth';
+import { validateUsername } from './signupRouter';
+import { createLuciaSessionCookie } from '../../../utils/session';
 
 export const passkeyRouter = router({
-  generateNewPasskeyChallenge: userProcedure
+  // generateNewPasskeyChallenge: userProcedure
+  //   .input(
+  //     z
+  //       .object({
+  //         authenticatorType: z.enum(['platform', 'cross-platform'])
+  //       })
+  //       .strict()
+  //   )
+  //   .query(async ({ ctx, input }) => {
+  //     const { db, user } = ctx;
+
+  //     const userId = user.id;
+  //     // Update to use read replicas when implemented - primary db
+  //     const userResponse = await db.query.users.findFirst({
+  //       where: eq(users.id, userId),
+  //       columns: {
+  //         id: true,
+  //         publicId: true,
+  //         username: true
+  //       }
+  //     });
+
+  //     if (!userResponse) {
+  //       throw new TRPCError({
+  //         code: 'BAD_REQUEST',
+  //         message: 'User not found'
+  //       });
+  //     }
+
+  //     const authenticatorAttachment = input.authenticatorType;
+
+  //     const passkeyOptions = await usePasskeys.generateRegistrationOptions({
+  //       userId: userId,
+  //       userPublicId: userResponse.publicId,
+  //       userName: userResponse.username,
+  //       userDisplayName: userResponse.username,
+  //       authenticatorAttachment: authenticatorAttachment
+  //     });
+
+  //     return { options: passkeyOptions };
+  //   }),
+
+  signUpWithPasskeyStart: limitedProcedure
     .input(
-      z
-        .object({
-          canUsePasskeyDirect: z.boolean()
-        })
-        .strict()
+      z.object({
+        username: zodSchemas.username(),
+        authenticatorType: z.enum(['platform', 'cross-platform'])
+      })
     )
-    .query(async ({ ctx, input }) => {
-      const { db, user } = ctx;
-
-      const userId = user.id;
-      // Update to use read replicas when implemented - primary db
-      const userResponse = await db.query.users.findFirst({
-        where: eq(users.id, userId),
-        columns: {
-          id: true,
-          publicId: true,
-          username: true
-        }
+    .query(async ({ input }) => {
+      const { username, authenticatorType } = input;
+      const publicId = nanoId();
```

Comment:
theres a possibility for a nanoId conflict here
not sure how we want to handle it

Reply:
make publicId a unique field, the chances of colliding is really low. So on the rare case it happens the query will throw error.